handler.go

```diff
@@ -449,12 +449,27 @@ func (h *Handler) handleFetch(ctx *Context, req *protocol.FetchRequest) *protoco
 					log.Printf("topic %s message provider error: %v", fetchTopic.Topic, err)
 				}
 
-				b, err := protocol.Encode(&protocol.MessageSet{
+				ms := protocol.MessageSet{
 					Offset: offset,
 					Message: &protocol.Message{
 						Value: data,
 					},
-				})
+				}
+
+				// Use the v1 message format for Fetch v2 or later. Fetch v2 supports either the v0
+				// or v1 message format, but for simplicitly we will always simulate a v1 message on
+				// v2+ protocols. See
+				// https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-Messagesets.
+				if req.APIVersion >= 2 {
+					// 0 is for v0 message format; 1 is for v1 message format.
+					ms.Message.MagicByte = 1
+					ms.Message.Timestamp = time.Now()
```

Comment:
No action now, but I just wanted to check my understanding. It seems like the `Timestamp` would be taken from the Flow messages once we wire that up. Does that sound right to you?

Reply:
Yep, with some consideration for how Flow's concepts of timestamps mesh with Kafka's, which I'm not 100% sure of presently but will hopefully be straightforward. For now this is for mocking out something approximating how that should look.